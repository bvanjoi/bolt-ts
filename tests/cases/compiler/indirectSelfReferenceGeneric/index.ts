// From `github.com/microsoft/TypeScript/blob/v5.7.2/tests/cases/compiler/indirectSelfReferenceGeneric.ts`, Apache-2.0 License

class a<T> extends b<T> { }
//~^ ERROR: Class 'b' used before its declaration.
//~| ERROR: 'a' is referenced directly or indirectly in its own base expression.
class b<T> extends a<T> { }
//~^ ERROR: 'b' is referenced directly or indirectly in its own base expression.
