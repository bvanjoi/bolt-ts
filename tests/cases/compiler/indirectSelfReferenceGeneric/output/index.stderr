  × Class 'b' used before its declaration.
   ╭─[../../tests/cases/compiler/indirectSelfReferenceGeneric/index.ts:3:20]
 3 │ b
   · ─
   ╰────

Advice: 
  ☞ Class 'b' is defined here
   ╭─[../../tests/cases/compiler/indirectSelfReferenceGeneric/index.ts:6:7]
 6 │ b
   · ─
   ╰────

  × 'b' is referenced directly or indirectly in its own base expression.
   ╭─[../../tests/cases/compiler/indirectSelfReferenceGeneric/index.ts:6:1]
 6 │ class b<T> extends a<T> { }
   · ───────────────────────────
   ╰────

  × 'a' is referenced directly or indirectly in its own base expression.
   ╭─[../../tests/cases/compiler/indirectSelfReferenceGeneric/index.ts:3:1]
 3 │ class a<T> extends b<T> { }
   · ───────────────────────────
   ╰────
